### PR TITLE
chore: simplify dependencies for the user

### PR DESCRIPTION
**Pull request by callicles**
* Use the new package for moose internal dependencies
* Update all internal projects to use the new moose lib
* Add a type helper for data model config

### CODE REVIEW

File: packages/ts-moose-lib/src/index.ts [hunk 1]
```diff
@@ -4,3 +4,18 @@ export interface Aggregation {
 }
 
 export type Key<T extends string | number | Date> = T;
+
+export enum IngestionFormat {
+  JSON = "JSON",
+  JSON_ARRAY = "JSON_ARRAY",
+}
+
+export type DataModelConfig = Partial<{
```
Comment (phiSgr) [inline packages/ts-moose-lib/src/index.ts]: What about making it generic, and the order_by has to be field names of the type parameter?

Comment (callicles) [reply packages/ts-moose-lib/src/index.ts]: Can you spell it out? I don't see what you are suggesting

Comment (phiSgr) [reply packages/ts-moose-lib/src/index.ts]: `order_by_fields?: (keyof T)[];`

And if I mistype the field name:
```
app/datamodels/models.ts:47:23 - error TS2820: Type '"observedTimestam"' is not assignable to type 'keyof Log'. Did you mean '"observedTimestamp"'?

47     order_by_fields: ["observedTimestam"],
```

But then this is a bit complicated if we want to support nested keys.

Comment (callicles) [reply packages/ts-moose-lib/src/index.ts]: We can start with that and then iterate

**Review by phiSgr — APPROVED** [review]

**Pull request merged**